### PR TITLE
ports core-components to react-hookz..

### PR DESCRIPTION
**Pull request by punkle**
## Hey, I just made a Pull Request!

ports core-components to react-hookz as far as possible.

There are a couple of hooks that have not been implemented in react-hookz.

 - useWindowSize, useCopyToClipboard, useObservable

I have not changed these. I am not very familiar with the debounce pattern
so I hope I have ported that correctly.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/core-components/src/hooks/useQueryParamState.ts [hunk 1]
```diff
@@ -75,7 +75,7 @@ export function useQueryParamState<T>(
     );
   }, [searchParamsString, setQueryParamState, stateName]);
 
-  useDebounce(
+  useDebouncedCallback(
```
Comment (Rugvip) [inline packages/core-components/src/hooks/useQueryParamState.ts:78]: I believe this should be `useDebouncedEffect` instead? Right now I don't think this code ends up doing much as the returned callback isn't called

**Review by Rugvip — COMMENTED** [review]
Thank you! 👍 

Looks good except I think the debounce bit needs a tweak

File: packages/core-components/src/hooks/useQueryParamState.ts (hunk 1 above)
Comment (punkle) [reply packages/core-components/src/hooks/useQueryParamState.ts]: Ive changed it now. The hook `useQueryParamState` seems to be only used by a deprecated LegacySearchPage, although it is also exported so perhaps its still used outside of the LegacySearchPage.

**Review by Rugvip — APPROVED** [review]
Alright, thank you, let's :shipit: 😁

**Pull request merged**